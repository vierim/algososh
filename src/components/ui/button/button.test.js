import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from './button';

const buttonText = 'Кнопка с текстом';

describe('Тестирование компонента Button', () => {
  it('Рендер кнопки с текстом', () => {
    const tree = renderer
      .create(<Button text={buttonText} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Рендер кнопки без текста', () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Рендер заблокированной кнопки', () => {
    const tree = renderer
      .create(<Button disabled={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Рендер кнопки с индикацией загрузки', () => {
    const tree = renderer
      .create(<Button isLoader={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Вызов колбека по клику на кнопку', () => {
    const fn = jest.fn();
    
    render(<Button text={buttonText} onClick={fn} />);

    const button = screen.getByText(buttonText);
    fireEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });
});
