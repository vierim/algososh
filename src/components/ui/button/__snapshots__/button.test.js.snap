// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button Рендер заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Рендер кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Рендер кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button Рендер кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка с текстом
  </p>
</button>
`;
